sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_custom_form_client.py

```diff
@@ -0,0 +1,296 @@
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from azure.core.exceptions import HttpResponseError
+from ._generated.models import AnalyzeOperationResult, Model
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._generated.models import TrainRequest, TrainSourceFilter
+from ._base_client import FormRecognizerClientBase
+from ._response_handlers import (
+    prepare_unlabeled_result,
+    prepare_labeled_result,
+)
+from ._helpers import get_pipeline_response, get_content_type, POLLING_INTERVAL
+from ._models import (
+    ModelInfo,
+    ModelsSummary,
+    CustomModel,
+    CustomLabeledModel,
+)
+if TYPE_CHECKING:
+    from ._credential import FormRecognizerApiKeyCredential
+
+
+class CustomFormClient(FormRecognizerClientBase):
+    """CustomFormClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of FormRecognizerApiKeyCredential if using an API key.
+    :type credential: ~azure.ai.formrecognizer.FormRecognizerApiKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
+        # type: (str, FormRecognizerApiKeyCredential, Any) -> None
+        super(CustomFormClient, self).__init__(credential=credential, **kwargs)
+        self._client = FormRecognizer(
+            endpoint=endpoint, credential=credential, pipeline=self._pipeline
+        )
+
+    @distributed_trace
+    def begin_training(self, source, source_prefix_filter="", include_sub_folders=False, **kwargs):
+        # type: (str, Optional[str], Optional[bool], Any) -> LROPoller
+        """Create and train a custom model. The request must include a source parameter that is an
+        externally accessible Azure storage blob container Uri (preferably a Shared Access Signature Uri).
+        Models are trained using documents that are of the following content type - 'application/pdf',
+        'image/jpeg', 'image/png', 'image/tiff'. Other type of content is ignored.
+
+        :param str source: An Azure Storage blob container URI.
+        :param str source_prefix_filter: A case-sensitive prefix string to filter documents in the source path for
+            training. For example, when using a Azure storage blob Uri, use the prefix to restrict sub
+            folders for training.
+        :param bool include_sub_folders: A flag to indicate if sub folders within the set of prefix folders
+            will also need to be included when searching for content to be preprocessed.
+        :return: LROPoller
+        :rtype: ~azure.core.polling.LROPoller
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        content_type = kwargs.pop("content_type", "application/json")
```

Comment:
Don't think content_type should default to "application/json" if you mention that it's included in the content type that is ignored. Can you use `get_content_type` here?

Reply:
discussed offline